code/clue.js

```diff
@@ -1,59 +1,342 @@
+
+  // This function does randomly select a favourite weapon.
+  shuffleFavouriteWeapon = () => {
+    mystery.favouriteWeapon = randomSelector(weapons); 
+  };
+  shuffleFavouriteWeapon();
+
+  // This will change the background color of the card to the one connected to the chosen killer and show the different properties of the killer. 
+  document.getElementById('killerCard').style.background = mystery.killer.color;
+  document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`;
+  document.getElementById('killerAge').innerHTML = `Age: ${mystery.killer.age}`;
+  document.getElementById('killerOccupation').innerHTML = `Occupation: ${mystery.killer.occupation}`;
+  killerImage.src = mystery.killer.image;
+  document.getElementById('killerDescription').innerHTML = `Description: ${mystery.killer.description}`;
+  document.getElementById('favouriteWeapon').innerHTML = `Favourite weapon: ${mystery.favouriteWeapon.name}`;
+
+  document.getElementById('killerCard').onclick = realPickKiller;
+};
+
+// PICK-WEAPON-FUNCTION! This function will be invoked when you click on the weapon card.- in the similar way as pickKiller.
+const pickWeapon = () => { // Shows loading animation
+  document.getElementById('loader2').style.display = 'block';
+  setTimeout(realPickWeapon, 1000); // Delays the real pickWeapon function and calls it after 1.5 sec.
+};
+document.getElementById('weaponCard').addEventListener('click', pickWeapon);
+
+const realPickWeapon = () => { // The REAL function getting the values for the weapons
+  document.getElementById('loader2').style.display = 'none';
+  document.getElementById('weaponCard').removeEventListener('click', pickWeapon); // So the loader doesn't pops up in next click, it looks a bit weird if that happens
+
+  mystery.weapon = randomSelector(weapons); // Randomly selects a weapon
+      
+
+  document.getElementById('weaponName').innerHTML = `${mystery.weapon.name}`;
+  document.getElementById('weaponWeight').innerHTML = `Weight: ${mystery.weapon.weight} pounds`;
+  weaponImage.src = mystery.weapon.image;
+
+  document.getElementById('weaponCard').onclick = realPickWeapon;
+};
+
+// PICK-ROOM-FUNCTION! This function will be invoked when you click on the room card.- in the similar way as pickKiller.
+const pickRoom = () => { // Shows loading animation
+  document.getElementById('loader3').style.display = 'block';
+  
+  setTimeout(realPickRoom, 1000); // Delays the real pickRoom function and calls it after 1.5 sec.
+};
+document.getElementById('roomCard').addEventListener('click', pickRoom);
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+const realPickRoom = () => { // The REAL function getting the values for the rooms
+  document.getElementById('loader3').style.display = 'none';
+  document.getElementById('roomCard').removeEventListener('click', pickRoom); // So the loader doesn't pops up in next click
```

Comment:
WOW!!! The loader-function is amazing! Great job with getting that to work so well